middleware/validateBody.js

```
@@ -14,14 +14,27 @@ const createError = require('http-errors');
 function validateBody(schema) {
     return (req, res, next) => {
         try {
-            // Check for empty body
-            if (Object.keys(req.body).length === 0) {
+            // For partial updates (like PATCH or PUT with partial data),
+            // we don't require a non-empty body
+            const isPartialSchema = schema._def && schema._def.description && 
+                schema._def.description.includes('partial');
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)